### DISCUSSION TITLE
Option to only scan Active Flows

### DISCUSSION THREAD

**Discussion by k-capehart** [Flow-Scanner/lightning-flow-scanner#89] [Ideas]
I love this plugin! It's very helpful when enforcing best practices in flows, something that is notoriously hard to do.

However, I don't see an easy way to only scan active flows. There might be situations where I deactivate a problematic flow in my org, but I don't want to delete it from my source control files quite yet. It would be great to be able to have a `--active-only` flag or something similar so that it only scans flows actively being used. 

I looked through the code to see if it would be a simple contribution, but it's probably simpler to ask here first in case I'm missing something in the configuration options.

**Comment by RubenHalman** [comment]
@k-capehart 
Thank you. That is a great idea, I think we can look at the status of the flow in the xml directly. Perhaps you can have a look at the core Scan function, as ideally I think we should have this in sfdx and vsce.(Maybe in the ruleconfig file, what do you think?) I think we should convert it to a ticket so we can collab on this, please do let us know how you would like to proceed with this!

**Reply by k-capehart** [reply]
@RubenHalman Sure, I'll see if I have some time this week or next to dive into it together. 

Also, I thought about it some more. Would you want the scanner to either:
- default to only scan active flows, with an option to include inactive flows?
- Or default to all flows (current functionality), with an option to exclude inactive flows?

I don't really have a preference from a user perspective but I think its worth thinking about.

**Reply by RubenHalman** [reply]
@k-capehart your question makes sense and the more i think about this the more i wonder: Wouldn't it be Best Practice NOT to have any inactive flows in your repository(Although i understand that is often unrealistic). If that would be the case, and we would want to potentially create a rule for it, i think by default all flows should be scanned(also inactive). hope this makes sense. 

@fkramer690 Do you happen to have any thoughts on this?

**Reply by k-capehart** [reply]
@RubenHalman I took a look at this and decided a possible good first step would be to create rule for Inactive Flows. I took a stab at it in this PR: https://github.com/Lightning-Flow-Scanner/lightning-flow-scanner-core/pull/83

Let me know what you think!

**Reply by RubenHalman** [reply]
@k-capehart Thank you, I think this is a great first step and this will help us further explore the topic within the community. Please allow me a few days to have your PR is included in the vscode and sfdx apps. If you want to explore this topic further, we might be able to pass a configuration similar to severity, ie. something called: 'included statusses'.  

However looking at your use case, I think personally I would just dump those flows in a different sub directory, or specify the flows in the exception configurations so they are not included in the scan. I am wondering if this might be a viable workable work around for you. Let us know, and thank you for joining us!

**Reply by k-capehart** [reply]
I think that will work for now. I was thinking of including the sfdx tool as part of my CI process, so we can enforce best practices before a PR is merged. I've worked in some older orgs with lots of flows, so a lot of them are inactive and it wouldn't make sense to include them in the scan. This will at least indicate to me which flows can be safely removed from the source control and scheduled for deletion so that I don't have to deal with a long list of violations for flows not being used. 

I appreciate it!
